### PR TITLE
[stable/redis] WIP: add init-container for set chown redis data dir 

### PR DESCRIPTION
**Pull request by LuckySB**
when securityContext and persistence claim template is enabled

**What this PR does / why we need it**:
with .Values.master.securityContext.enabled Redis run as non privileged user, but some persistence volume provisioners (such as cephfs) mount volume with owner root, so redis cannot create backup files - appendonly.aof or dump.rdb, and crashed

### CODE REVIEW

File: stable/redis/templates/redis-master-statefulset.yaml [hunk 1]
```diff
@@ -135,6 +135,16 @@ spec:
           mountPath: /opt/bitnami/redis/etc/redis.conf
           subPath: redis.conf
         {{- end }}
+      {{- if and .Values.master.persistence.enabled (not .Values.persistence.existingClaim) | and .Values.master.securityContext.enabled }}
+      initContainers:
+      - name: redis-init
```
Comment (juan131) [inline stable/redis/templates/redis-master-statefulset.yaml]: I'd use a name such as `volume-permissions`

File: stable/redis/templates/redis-master-statefulset.yaml [hunk 2]
```diff
@@ -135,6 +135,16 @@ spec:
...
+        image: alpine:3.8
```
Comment (juan131) [inline stable/redis/templates/redis-master-statefulset.yaml]: This should be configurable depending on the `values.yaml`:
```
        image: "{{ .Values.volumePermissions.image.name }}:{{ .Values.volumePermissions.image.tag }}"
        imagePullPolicy: {{ .Values.volumePermissions.image.pullPolicy | quote }}
```

File: stable/redis/templates/redis-master-statefulset.yaml [hunk 3]
```diff
@@ -135,6 +135,16 @@ spec:
           mountPath: /opt/bitnami/redis/etc/redis.conf
           subPath: redis.conf
         {{- end }}
+      {{- if and .Values.master.persistence.enabled (not .Values.persistence.existingClaim) | and .Values.master.securityContext.enabled }}
```
Comment (juan131) [inline stable/redis/templates/redis-master-statefulset.yaml]: Why do you need the pipe?

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the PR. Please take a look to my comments.

File: stable/redis/templates/redis-master-statefulset.yaml (hunk 2 above)
Comment (LuckySB) [reply stable/redis/templates/redis-master-statefulset.yaml]: other image structure do not have filed "name", but have "repository"
and use template helpers

File: stable/redis/templates/redis-master-statefulset.yaml (hunk 1 above)
Comment (LuckySB) [reply stable/redis/templates/redis-master-statefulset.yaml]: ok

File: stable/redis/templates/redis-master-statefulset.yaml (hunk 3 above)
Comment (LuckySB) [reply stable/redis/templates/redis-master-statefulset.yaml]: i am no so familiar with go template
reworked

**Comment by LuckySB** [comment]
@juan131 i am sorry, by mistake I forgot to sign a commit
I corrected all your requests, but did not change the commit

I made a new PR https://github.com/helm/charts/pull/7928, please look at it

this commit i closed

**Pull request closed without merging**